src/core/Akka.Streams/Dsl/FlowWithContext.cs

```diff
@@ -17,28 +18,27 @@ namespace Akka.Streams.Dsl
     /// operations.
     /// 
     /// An "empty" flow can be created by calling <see cref="FlowWithContext.Create{TCtx,TIn}"/>.
-    /// 
-    /// API MAY CHANGE
-    ///</summary> 
-    public sealed class FlowWithContext<TCtxIn, TIn, TCtxOut, TOut, TMat>
+    ///</summary>
+    [ApiMayChange]
+    public sealed class FlowWithContext<TIn, TCtxIn, TOut, TCtxOut, TMat>
         : GraphDelegate<FlowShape<(TIn, TCtxIn), (TOut, TCtxOut)>, TMat>
     {
-        internal FlowWithContext(Flow<(TIn, TCtxIn), (TOut, TCtxOut), TMat> flow) 
+        internal FlowWithContext(Flow<(TIn, TCtxIn), (TOut, TCtxOut), TMat> flow)
             : base(flow)
         {
         }
-        
+
         ///<summary>
         /// Transform this flow by the regular flow. The given flow must support manual context propagation by
         /// taking and producing tuples of (data, context).
         /// 
         /// This can be used as an escape hatch for operations that are not (yet) provided with automatic
         /// context propagation here.
         ///</summary>
-        public FlowWithContext<TCtxIn, TIn, TCtx2, TOut2, TMat> Via<TCtx2, TOut2, TMat2>(
+        public FlowWithContext<TIn, TCtxIn, TOut2, TCtx2, TMat> Via<TOut2, TCtx2, TMat2>(
```

Comment:
Instead of changing the existing APIs, can we make it call the new API? We can adapt the existing API @ismaelhamed ? 

This applies to all other API changes. We can deprecate the old ones and have it call the new ones, that is, old over new. Is that possible?